drivers/SmartThings/zwave-thermostat/src/ct100-thermostat/init.lua

```diff
@@ -237,19 +98,17 @@ local function thermostat_mode_report_handler(self, device, cmd)
     event = capabilities.thermostatMode.thermostatMode.emergency_heat()
   end
 
-  device:set_field(MODE, mode, {persist = true})
-
   if (event ~= nil) then
     device:emit_event(event)
   end
 
-  local heating_setpoint = device:get_field(CURRENT_HEATING_SETPOINT)
-  local cooling_setpoint = device:get_field(CURRENT_COOLING_SETPOINT)
-  local current_temperature = device:get_field(TEMPERATURE)
+  local heating_setpoint = device:get_latest_state("main", capabilities.thermostatHeatingSetpoint.ID, capabilities.thermostatHeatingSetpoint.heatingSetpoint.NAME, 0)
+  local cooling_setpoint = device:get_latest_state("main", capabilities.thermostatCoolingSetpoint.ID, capabilities.thermostatCoolingSetpoint.coolingSetpoint.NAME, 0)
+  local current_temperature = device:get_latest_state("main", capabilities.temperatureMeasurement.ID, capabilities.temperatureMeasurement.temperature.NAME, 0)
 
   device:send(ThermostatOperatingState:Get({}))
   if mode == ThermostatMode.mode.COOL or
-    ((mode == ThermostatMode.mode.COOL or mode == ThermostatMode.mode.OFF) and (current_temperature > (heating_setpoint + cooling_setpoint) / 2)) then
+    ((mode == ThermostatMode.mode.AUTO or mode == ThermostatMode.mode.OFF) and (current_temperature > (heating_setpoint + cooling_setpoint) / 2)) then
```

Comment:
Why do we need to read the setpoints in the `((mode == ThermostatMode.mode.AUTO or mode == ThermostatMode.mode.OFF) and (current_temperature > (heating_setpoint + cooling_setpoint) / 2))` case? I dont understand the `current_temperature > (heating_setpoint + cooling_setpoint) / 2)` at all.

Reply:
me neither, but the DTH does it

```
	if (mode == "cool") {
		thermostatSetpoint = coolingSetpoint
	} else if (mode == "auto" || mode == "off") {
		// Set thermostatSetpoint to the setpoint closest to the current temperature
		def currentTemperature = getTempInLocalScale("temperature")
		if (currentTemperature > (heatingSetpoint + coolingSetpoint)/2) {
			thermostatSetpoint = coolingSetpoint
		}
	}
```